print_docs.py

```diff
@@ -268,6 +268,18 @@ def html_head(title):
         <link rel="shortcut icon" href="https://leanprover-community.github.io/assets/img/lean.ico">
         <title>mathlib docs: {1}</title>
         <meta charset="UTF-8">
+        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
+        <script>
+        MathJax = {{
+          tex: {{
+            inlineMath: [['$', '$'], ['\\(', '\\)']]
```

Comment:
IIRC Zulip (and other LaTeX/makdown combos?) don't use single `$` delimiters for inline MathJax. It results in a lot of false positives. I'm not sure how big a worry it is here. Is MathJax smart enough to avoid inserting itself in code blocks `f $ g $ h a`? If not, double `$$` is safer.

Reply:
Is there a way to see how the docs render with this patch in place? I.e., can we test what happens?

Reply:
http://robertylewis.com/mathlib_docs/ is a build with @glangmead 's changes enabled.

Reply:
Something funny is going on [here](https://robertylewis.com/mathlib_docs/conv.html). I don't think the parens are escaped in the markdown. Does ['\\(', '\\)'] mean that everything in parens will become Mathjax??

Reply:
It *should* mean that `\( x + \sqrt{y + z} \)` is parsed as mathjax. So the `\` has to be explicitly in the source before the parens.

Reply:
I mean, that's what I assumed, but not what's happening.

Reply:
I see the problem with conv. I'll look into that, we can't have false positives. I note that conv starts life as a markdown doc, maybe there's something different about that code path. I'll try to break a .lean file in the same way to check.

Reply:
It seems to be happening with everything that's surrounded by parens: https://robertylewis.com/mathlib_docs/geometry/manifold/manifold.html

Reply:
Agreed. Probably the parens in my configuration array need more backslashes. On the other hand if we're removing $ then I think we can accept the defaults and I can remove that line. I'll try both so that I understand what's going on.

Note that double-dollar and backslash-bracket will switch to display mode, which may not be what zulip is doing. That makes backslash-paren the only way to do inline math from a lean file. I'll see if MathJax has a way to exclude its processing from regions of the doc so we could have the best of both worlds and support dollars in the comments.

Reply:
Gosh is there an existing bug with parens? See https://leanprover-community.github.io/mathlib_docs/geometry/manifold/real_instances.html where the definition of euclidean_space in lean is `def euclidean_space (n : ℕ) : Type := (fin n → ℝ)` but on the web some of it is deleted: `def euclidean_space : ℕ → Type`.

Reply:
MathJax does exclude itself from `<code>` blocks and you can configure it to exclude more css classes. I'll test that out.

Reply:
For what it's worth, MathJax is already smart enough to avoid running in `<pre>` or `<code>` blocks. So `$` occurrences in Lean code shouldn't trigger it, unless someone has forgotten to put backticks around it. So maybe the simple solution is to keep single `$` and remove `\(` as delimiters. I doubt there are many if any plain text lines in mathlib that have two `$`s.

Reply:
> Gosh is there an existing bug with parens? See https://leanprover-community.github.io/mathlib_docs/geometry/manifold/real_instances.html where the definition of euclidean_space in lean is `def euclidean_space (n : ℕ) : Type := (fin n → ℝ)` but on the web some of it is deleted: `def euclidean_space : ℕ → Type`.

This isn't a bug. The web docs are only displaying the type, which is `ℕ → Type`. The `ℕ` argument doesn't have to be bound to print `Type`, but in the source it's referenced in the body of the definition `(fin n → ℝ)`.

Reply:
It is maybe a badly phrased doc string though. It should "informally bind" `n`, i.e. "`euclidean_space n` is the space ℝ^n."

Reply:
I learned a few things and will update the PR shortly. First, I've swung back to wanting dollar signs and NOT the parens or brackets, since the latter do not pass cleanly through markdown. Stackoverflow has the same problem, see https://math.meta.stackexchange.com/a/18714. So: only single and double dollars should be processed, which will be what most people guess they should use anyway. 

Further, I've added all the lean-related css classes Rob defined in the python script as exclusions for MathJax. Some of those do not also have pre or code around them, so this should help keep MathJax out of the Lean stuff entirely. It can be further tweaked later of course!